platform/macos/src/MGLMapView.mm

```diff
@@ -717,11 +704,6 @@ - (BOOL)wantsBestResolutionOpenGLSurface {
     return !_isTargetingInterfaceBuilder;
 }
 
-- (CGLContextObj)context {
-    MGLOpenGLLayer *layer = _isTargetingInterfaceBuilder ? nil : (MGLOpenGLLayer *)self.layer;
-    return layer.openGLContext.CGLContextObj;
```

Comment:
`MGLOpenGLStyleLayer.context` depends on this code and will crash getting this property, which is still declared in MGLMapView_Private.h.

Reply:
i.e. we should remove it from the header file too?

Reply:
As I mentioned in https://github.com/mapbox/mapbox-gl-native/pull/14763#pullrequestreview-241939708, I’d prefer if we keep `MGLOpenGLStyleLayer.context` working, which would require bringing this code back or working it into MGLMapView+OpenGL.mm somehow. The code is quite harmless – all it depends on is the fact that `layer` happens to be an NSOpenGLLayer. If we need to relax that assumption, then we should keep this code around but have it return `nil` if `layer` isn’t an NSOpenGLLayer.

Reply:
Is there a test for this? I was running unit tests, and was running the custom OpenGL layer manually in the test app, and couldn't provoke a test. If removing this function breaks the code, a test case should catch that.

Reply:
Looking further into this, it looks like this code isn't used at all, anywhere, in our codebase. Can we just remove it?

Reply:
`MGLOpenGLStyleLayer.context` is part of the undocumented but public interface of that method. I’m using this method for a project but don’t know who else is at the moment.

You’re right that we should have more tests around custom layers at the SDK level. We skimped on that originally as part of keeping quiet about the feature, but in hindsight that was unnecessary. Still, a method declaration without a corresponding implementation should at least trigger a static analyzer warning.